pkg/wallet/account/account.go

```diff
@@ -0,0 +1,246 @@
+package account
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"errors"
+	"fmt"
+
+	"github.com/awnumar/memguard"
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/massalabs/station-massa-wallet/pkg/crypto"
+	"github.com/massalabs/station-massa-wallet/pkg/types"
+	"github.com/massalabs/station-massa-wallet/pkg/types/object"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	AccountLastVersion    = 1
+	AccountUnknownVersion = 0
+)
+
+var (
+	ErrInvalidPrivateKey = errors.New("invalid private key")
+	ErrGeneratingKeyPair = errors.New("generating ed25519 key-pair")
```

Comment:
This error is too specific and will make your code more complex to read in the error handling section.
I suggest to do the following:
```golang
var ErrRuntime = errors.New("runtime execution")
...
publicKeyBytes, privateKeyBytes, err := ed25519.GenerateKey(rand.Reader)
if err != nil {
	return nil, fmt.Errorf("key generation failed: %w: %w", ErrRuntime, err)
}
...
```
That way you'll be able to catch the ErrRuntime to return http code 500.

Note: the second %w is optional, you could use %v instead, but doing it you wrap the underlying error, exposing it to the caller if at somepoint he will need to differenciate different errors.